### PR TITLE
chore(async-migrations): Make missing migrations error noticeable

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

When a Python process was stopped due to an invalid state of async migrations (e.g. a required migration must have been ran on an earlier version), it was not obvious at all that that's what happened. Case in point:

hmm

It took quite a back-and-forth between us and a user who had this issue just to find the critical logged message ([see Slack thread](https://posthogusers.slack.com/archives/C01GLBKHKQT/p1657143298843989)).

## Changes

Show-stopping issues are now obvious, by using `print_warning()` instead of plain `print()` (+ some stronger wording):

ohh

### CODE REVIEW

File: posthog/models/async_migration.py [hunk 1]
```diff
@@ -49,6 +49,11 @@ class Meta:
 
     parameters: models.JSONField = models.JSONField(default=dict)
 
+    def __str__(self) -> str:
```
Comment (tiina303) [inline posthog/models/async_migration.py:52]: maybe make it `get_version_requirements` or something like that, feels a bit odd for `__str__` to have this

**Review by tiina303 — CHANGES_REQUESTED** [review]
So the output in the end I see in your summary is confusing. 0005 isn't required to be completed yet

**Comment by Twixes** [comment]
Yeah, for the screenshot I just used `AsyncMigration.objects.all()`. This PR doesn't actually change `get_necessary_migrations()` logic.

File: posthog/models/async_migration.py (hunk 1 above)
Comment (Twixes) [reply posthog/models/async_migration.py]: Fair point, updated.

**Review by tiina303 — APPROVED** [review]
👍

**Pull request merged**